src/ShellRouting/IShellRouter.cs

```diff
@@ -26,6 +27,16 @@ public interface IShellHandler
         ///     The incoming message to be handled.
         /// </param>
         public void Handle(Message message);
```

Comment:
Why not to just have an async version of the message handling?
Less execution branches to maintain and test...

Reply:
Sounds good, will remove this, especially since it's a breaking change already.